ios/src/TangramMap/TGMapViewController.h

```diff
@@ -0,0 +1,731 @@
+ Some already made basemap styles can be found
+ <a href="https://mapzen.com/documentation/cartography/styles/">here</a> using Mapzen
+ data sources.
+
+ To use basemap styles you can <a href="https://mapzen.com/developers/sign_in">sign up for
+ an API key</a> and load it through your application:
+
+ ```swift
+ let sceneURL = "https://mapzen.com/carto/walkabout-style-more-labels/walkabout-style-more-labels.yaml";
+ view.loadSceneFile(sceneURL, sceneUpdates: [ TGSceneUpdate(path: "sources.mapzen.url_params", value: "{ api_key: \(YOUR_API_KEY) }") ]);
+ ```
+ */
+@interface TGMapViewController : GLKViewController <UIGestureRecognizerDelegate>
+
+/**
+ If continuous rendering is set to `true`, the map view will render continuously, otherwise,
+ the map will render only when necessary.
+ Some styles can be set to be `animated` and this default value will be set appropriately,
+ see the <a href="https://mapzen.com/documentation/tangram/scene/#animated">
+ style parameter</a> for more details.
+
+ @note Any changes to this value will override the default induced value from the `animated`
+ style parameter.
+ Before making changes to this value, make sure battery usage is not something critical
+ for your application.
+ */
+@property (assign, nonatomic) BOOL continuous;
+
+#pragma mark Delegates
+
+/// Assign a gesture recognizer delegate, may be `nil`, see `TGRecognizerDelegate` for more details
+@property (weak, nonatomic, nullable) id<TGRecognizerDelegate> gestureDelegate;
+/// Assign a map viw delegate, may be `nil`, see `TGMapViewDelegate` for more details
+@property (weak, nonatomic, nullable) id<TGMapViewDelegate> mapViewDelegate;
+
+#pragma mark Camera/View modifiers
+
+/* The following are computed property. They return sensible defaults when the above .map property is nil */
+
+/// Assign a `TGCameraType` to the view camera
+@property (assign, nonatomic) TGCameraType cameraType;
+/// Assign a longitude and latitude to the map view camera
+@property (assign, nonatomic) TGGeoPoint position;
+/// Assign a floating point zoom to the map view camera
+@property (assign, nonatomic) float zoom;
+/// Assign a rotation angle in radians to the map view camera
+@property (assign, nonatomic) float rotation;
+/// Assign a tilt angle in radians to the map view camera
+@property (assign, nonatomic) float tilt;
+
+/**
+ Assign `TGHttpHandler` for network request management.
+ A http handler with a default configuration is already provided to the map view.
+
+ Must be non-`nil`.
+
+ @note Assigning a http handler is optional and should only be done if you want to change
+ any network related configuration (cache path and network cache size, both memory and disk,
+ or be notified when a network request completes).
+ More informations on the default provided configuration can be found on the description
+ of `TGHttpHandler`.
+ */
+@property (strong, nonatomic) TGHttpHandler* httpHandler;
+
+#pragma mark Debug toggles
+
+/**
```

Comment:
Outside of scope - it might be nice to have a way to set an array of debug flags some day.